### PR TITLE
Add v0.6.0 release notes

### PR DESCRIPTION
**Pull request by leewujung**
As @lsetiawan pointed out it is easier to crowdsource our release notes with version control. 

This PR adds release notes for the upcoming v0.6.0. We can collaborate on this the PR to make it complete as we move forward on the tasks.

I also changed the file from .rst to .md to allow direct copy-pasting this into the actual release notes later.

### CODE REVIEW

File: docs/source/whats-new.md [hunk 1]
```diff
@@ -1,47 +1,88 @@
...
+- Add `water_level` to processed datasets (Sv, MVBS, TS) for convenient downstream conversion to depth (#259, #583, #615)
+- Add additional environment variables for EK80 data (#616)
+- Add missing platform data variables (#592, #631)
+
+## New features and other enhancements
+
+- Add parser for Echoview ECS file (#510)
+- Add provenance to raw-converted and processed datasets (#621)
+- Consolidate convention specs into a single yml file for pre-loading when creating `EchoData` objects (#565)
+
```
Comment (lsetiawan) [inline docs/source/whats-new.md:45]:
```suggestion

## CI improvements

- Upgrade python minimum to 3.8 and tidy up packaging (#604, #608, #609)
- Upgrade echopype development status to Beta (#610)
- Update setup-services.py to include images & volumes subtleties (#651)

```

Comment (lsetiawan) [inline docs/source/whats-new.md:45]:
```suggestion

## CI improvements

- Upgrade python minimum to 3.8 and tidy up packaging (#604, #608, #609)
- Upgrade echopype development status to Beta (#610)
- Update setup-services.py to include images & volumes subtleties (#651)

```

File: docs/source/whats-new.md [hunk 2]
```diff
@@ -1,47 +1,94 @@
...
+## CI improvements
+
+- Upgrade python minimum to 3.8 and tidy up packaging (#604, #608, #609)
+- Upgrade echopype development status to Beta (#610)
+- Update setup-services.py to include images & volumes subtleties (#651)
```
Comment (lsetiawan) [inline docs/source/whats-new.md:50]:
```suggestion
- Update `setup-services.py` to include images & volumes subtleties (#651)
```
There is some weird link on this one, so changing it to code.

**Comment by leewujung** [comment]
As a note, we can add in docs the final outcomes of all the changes (like dimensions and names) and where the discussions came from (in terms of specific comments in relevant issues, if those exist), and then link in the list here, so that it is easy for everybody including us what was done.

File: docs/source/whats-new.md [hunk 3]
```diff
@@ -1,47 +1,94 @@
...
+
+## Changes of netCDF data model
+
+- Move and rename the original `Beam` and `Beam_power` group to be subgroups under the `Sonar` group, in order to comply with the structure defined in the convention  (#567, #574, #605, #606, #611)
+  - `Beam` --> `Sonar/Beam_group1`: contains either raw power or power/angle data (for all sonar models other than EK80) or complex data (EK80)
+  - `Beam_power` --> `Sonar/Beam_group2`: contains power/angle when complex data occupies `Sonar/Beam_group1`; only exists for EK80 data when both power/angle data and complex data exist in the file
+- Rename the coordinate `range_bin` to `range_sample` to make it obvious that this coordinate indicates the digitization sample number for the sound waveform or intensity time series, and hence it takes a form of sequential integers 0, 1, 2, 3, ... (#595)
+- Rename the data variable `range` in the calibrated Sv or TS dataset to `echo_range`, so that it is not confused with the python reserved vocabulary (#590)
+- Rename the coordinate `quadrant` for EK80 data to `beam` (#619)
+- Add coordinate `beam` with length 1 for all sonar models (#638, #646)
```
Comment (b-reyes) [inline docs/source/whats-new.md:19]: Not a huge deal, but as stated in the final point, AD2CP is not included here. If we we want to change this, we could say something like: 

Add coordinate `beam` with length 1 for all sonar models, not inluding AD2CP (#638, #646)

File: docs/source/whats-new.md [hunk 4]
```diff
@@ -1,47 +1,94 @@
...
+- Rename the data variable `Sp` to `TS` since "point backscattering strength" is a Simrad terminology and target strength (TS) is clearly defined and widely used. (#615)
+- Rename time dimensions in the `Platform` group (`location_time`: `time1`, `mru_time`: `time2`) (#518, #631, #647)
+- Rename the coordinate `frequency` to `channel` for all groups, to be more flexible (can accommodate channels with identical frequencies #490) and reasonable (since for broadband data the channel frequency is only nominal #566) (ADD WHEN KNOWN)
+- Change `src_filenames` string attribute to `source_filenames` list-of-strings variable (#620, #621)
+- Note: the data structure of sonar model AD2CP remains unchanged in this release and will be re-organized in later releases.
+
```
Comment (b-reyes) [inline docs/source/whats-new.md:25]: We need to also add an item here for the addition of missing required variables in Platform groups (#592, #649)

File: docs/source/whats-new.md (hunk 3 above)
Comment (leewujung) [reply docs/source/whats-new.md]: I agree it is better here. Just changed.

File: docs/source/whats-new.md (hunk 4 above)
Comment (leewujung) [reply docs/source/whats-new.md:26]: this is listed in the section "Addition of attributes and variables in raw-converted and processed data" below.

File: docs/source/whats-new.md [hunk 5]
```diff
@@ -1,47 +1,95 @@
...
+- Rename the data variable `range` in the calibrated Sv or TS dataset to `echo_range`, so that it is not confused with the python reserved vocabulary (#590)
+- Rename the coordinate `quadrant` for EK80 data to `beam` (#619)
+- Add coordinate `beam` with length 1 for all sonar models, except for AD2CP (#638, #646)
+- Rename the data variable `Sp` to `TS` since "point backscattering strength" is a Simrad terminology and target strength (TS) is clearly defined and widely used. (#615)
+- Rename time dimensions in the `Platform` group (`location_time`: `time1`, `mru_time`: `time2`) (#518, #631, #647)
+- Rename the coordinate `frequency` to `channel` for all groups, to be more flexible (can accommodate channels with identical frequencies #490) and reasonable (since for broadband data the channel frequency is only nominal #566) (ADD WHEN KNOWN)
+- Rename the data variable `heave` to `vertical_offset` in the Platform group (#592, #623)
+- Change `src_filenames` string attribute to `source_filenames` list-of-strings variable (#620, #621)
+- Note: the data structure of sonar model AD2CP remains unchanged in this release and will be re-organized in later releases.
```
Comment (leewujung) [inline docs/source/whats-new.md:25]:
```suggestion
```
Remove this line since it is taken care of earlier.

File: docs/source/whats-new.md [hunk 6]
```diff
@@ -1,47 +1,94 @@
...
+See [GitHub releases page](https://github.com/OSOceanAcoustics/echopype/releases) for the complete history.
 
-v0.5.6 (2022 Feb 10)
---------------------
+# v0.6.0 (2022 May 4)
 
-Overview
-~~~~~~~~
+## Overview
+This is a major release that contains changes that enhances the compliance of echopype data model (and hence generated file structure) to the [SONAR-netCDF4 ver.1 convention](http://www.ices.dk/sites/pub/Publication%20Reports/Cooperative%20Research%20Report%20(CRR)/CRR341.pdf) [LINK REQUIRES LOGIN??? CHECK AGAIN BEFORE RELEASE]. In addition, some variables were renamed to improve intuitive understanding of sonar data, provenance and standardized attributes are added to the processed dataset (e.g. Sv), the deprecated old API (<0.5.0) was removed, and some bugs were fixed.
```
Comment (lsetiawan) [inline docs/source/whats-new.md:9]:
```suggestion
This is a major release that contains changes that enhances the compliance of echopype data model (and hence generated file structure) to the [SONAR-netCDF4 convention Version 1.0 ](http://www.ices.dk/sites/pub/Publication%20Reports/Cooperative%20Research%20Report%20(CRR)/CRR341.pdf). In addition, some variables were renamed to improve intuitive understanding of sonar data, provenance and standardized attributes are added to the processed dataset (e.g. Sv), the deprecated old API (<0.5.0) was removed, and some bugs were fixed.
```

I checked the PDF url and I don't see any login requirements.

File: docs/source/whats-new.md [hunk 7]
```diff
@@ -1,47 +1,94 @@
 What's new
 ==========
 
-See `GitHub releases page <https://github.com/OSOceanAcoustics/echopype/releases>`_ for the complete history.
+See [GitHub releases page](https://github.com/OSOceanAcoustics/echopype/releases) for the complete history.
 
-v0.5.6 (2022 Feb 10)
---------------------
+# v0.6.0 (2022 May 4)
```
Comment (lsetiawan) [inline docs/source/whats-new.md:6]:
```suggestion
# v0.6.0 (2022 May 26)
```

File: docs/source/whats-new.md [hunk 8]
```diff
@@ -1,47 +1,94 @@
...
+- Rename the data variable `Sp` to `TS` since "point backscattering strength" is a Simrad terminology and target strength (TS) is clearly defined and widely used. (#615)
+- Rename time dimensions in the `Platform` group (`location_time`: `time1`, `mru_time`: `time2`) (#518, #631, #647)
+- Rename the coordinate `frequency` to `channel` for all groups, to be more flexible (can accommodate channels with identical frequencies #490) and reasonable (since for broadband data the channel frequency is only nominal #566) (ADD WHEN KNOWN)
+- Rename the data variable `heave` to `vertical_offset` in the Platform group (#592, #623)
+- Change `src_filenames` string attribute to `source_filenames` list-of-strings variable (#620, #621)
+
+
+## Changes of `EchoData` group access pattern
+
+- The groups can now be accessed via a path in the form `echodata["Sonar/Platform"]`, `echodata["Sonar/Platform"]`, `echodata["Sonar/Beam_groupX"]`, etc. using [DataTree](https://github.com/xarray-contrib/datatree) functionalities. (#611)
```
Comment (lsetiawan) [inline docs/source/whats-new.md:29]:
```suggestion
- The groups can now be accessed via a path in the form `echodata["Sonar/Platform"]`, `echodata["Sonar/Platform"]`, `echodata["Sonar/Beam_groupX"]`, etc. using [DataTree v0.0.4](https://github.com/xarray-contrib/datatree/tree/0.0.4) functionalities. (#611)
```

File: docs/source/whats-new.md [hunk 9]
```diff
@@ -1,47 +1,94 @@
...
+- Rename the coordinate `range_bin` to `range_sample` to make it obvious that this coordinate indicates the digitization sample number for the sound waveform or intensity time series, and hence it takes a form of sequential integers 0, 1, 2, 3, ... (#595)
+- Rename the data variable `range` in the calibrated Sv or TS dataset to `echo_range`, so that it is not confused with the python reserved vocabulary (#590)
+- Rename the coordinate `quadrant` for EK80 data to `beam` (#619)
+- Add coordinate `beam` with length 1 for all sonar models, except for AD2CP (#638, #646)
+- Rename the data variable `Sp` to `TS` since "point backscattering strength" is a Simrad terminology and target strength (TS) is clearly defined and widely used. (#615)
+- Rename time dimensions in the `Platform` group (`location_time`: `time1`, `mru_time`: `time2`) (#518, #631, #647)
+- Rename the coordinate `frequency` to `channel` for all groups, to be more flexible (can accommodate channels with identical frequencies #490) and reasonable (since for broadband data the channel frequency is only nominal #566) (ADD WHEN KNOWN)
```
Comment (lsetiawan) [inline docs/source/whats-new.md:22]: There's still a `(ADD WHEN KNOWN)` here. Maybe @b-reyes can list out the PR numbers related to this work?

**Comment by lsetiawan** [comment]
Okay I've gone through the whole release notes. I think this covered all of our work.. (at least the one I paid attention to 😆). There's only one thing to do for @b-reyes. Any other than that, feel free to finalize my suggestions by doing a commit batch. Thanks!

File: docs/source/whats-new.md [hunk 10]
```diff
@@ -1,47 +1,94 @@
...
+- Rename the data variable `range` in the calibrated Sv or TS dataset to `echo_range`, so that it is not confused with the python built-in function (#590)
+- Rename the coordinate `quadrant` for EK80 data to `beam` (#619)
+- Add coordinate `beam` with length 1 for all sonar models, except for AD2CP (#638, #646)
+- Rename the data variable `Sp` to `TS` since "point backscattering strength" is a Simrad terminology and target strength (TS) is clearly defined and widely used. (#615)
+- Rename time dimensions in the `Platform` group (`location_time`: `time1`, `mru_time`: `time2`) (#518, #631, #647)
+- Rename the coordinate `frequency` to `channel` for all groups, to be more flexible (can accommodate channels with identical frequencies #490) and reasonable (since for broadband data the channel frequency is only nominal #566) (#657)
+- Rename the data variable `heave` to `vertical_offset` in the Platform group (#592, #623)
+- Change `src_filenames` string attribute to `source_filenames` list-of-strings variable (#620, #621)
+Bring consistency to the names of the time coordinates for the `Platform` and `Environment groups` (#656, #672)
```
Comment (b-reyes) [inline docs/source/whats-new.md:25]:
```suggestion
- Bring consistency to the names of the time coordinates for the `Platform` and `Environment groups` (#656, #672)
```

**Pull request merged**